src/controllers/mainController.ts

```diff
@@ -189,18 +189,27 @@ export default class MainController implements vscode.Disposable {
                     return this._objectExplorerProvider.refreshNode(this._objectExplorerProvider.currentNode);
 
                 });
+
+                // initiate the scripting service
+                this._scriptingService = new ScriptingService(this._connectionMgr, this._vscodeWrapper);
                 this._context.subscriptions.push(
                     vscode.commands.registerCommand(
-                        Constants.cmdScriptSelect, async (treeNodeInfo: TreeNodeInfo) => {
-                    const objectNames = treeNodeInfo.label.split('.');
-                    const tableName = `[${escapeCharacters(objectNames[0])}].[${escapeCharacters(objectNames[1])}]`;
-                    const databaseName = `[${escapeCharacters(self.getDatabaseName(treeNodeInfo))}].`;
-                    const selectStatement = Constants.scriptSelectText + databaseName + tableName;
-                    self.onNewQuery(treeNodeInfo.sessionId, selectStatement).then((result) => {
-                        if (result) {
-                            self.onRunQuery();
-                        }
-                    });
+                        Constants.cmdScriptSelect, async (node: TreeNodeInfo) => {
+                    const uri = await this._untitledSqlDocumentService.newQuery();
+                    if (!this.connectionManager.isConnected(uri.toString())) {
+                        const connectionCreds = node.connectionCredentials;
+                        const databaseName = `${escapeCharacters(self.getDatabaseName(node))}`;
+                        node.connectionCredentials.database = databaseName;
+                        this._statusview.languageFlavorChanged(uri.toString(), Constants.mssqlProviderName);
+                        this._statusview.sqlCmdModeChanged(uri.toString(), false);
+                        await this.connectionManager.connect(uri.toString(), connectionCreds);
+                        const selectStatement = await this._scriptingService.scriptSelect(node, uri.toString());
+                        self.onNewQuery(node.sessionId, selectStatement).then((result) => {
+                            if (result) {
```

Comment:
This is already async - suggest just using await unless you have a specific reason for not wanting to wait for onNewQuery to complete